streamer/src/quic.rs

```diff
@@ -0,0 +1,188 @@
+use {
+    crossbeam_channel::Sender,
+    futures_util::stream::StreamExt,
+    pem::Pem,
+    pkcs8::{der::Document, AlgorithmIdentifier, ObjectIdentifier},
+    quinn::{Endpoint, EndpointConfig, ServerConfig},
+    rcgen::{CertificateParams, DistinguishedName, DnType, SanType},
+    solana_perf::packet::PacketBatch,
+    solana_sdk::{
+        packet::{Packet, PACKET_DATA_SIZE},
+        signature::Keypair,
+    },
+    std::{
+        error::Error,
+        net::{IpAddr, UdpSocket},
+        sync::{
+            atomic::{AtomicBool, Ordering},
+            Arc,
+        },
+        thread,
+        time::Duration,
+    },
+    tokio::{
+        runtime::{Builder, Runtime},
+        time::timeout,
+    },
+};
+
+/// Returns default server configuration along with its PEM certificate chain.
+#[allow(clippy::field_reassign_with_default)] // https://github.com/rust-lang/rust-clippy/issues/6527
+fn configure_server(
+    identity_keypair: &Keypair,
+    gossip_host: IpAddr,
+) -> Result<(ServerConfig, String), Box<dyn Error>> {
+    let (cert_chain, priv_key) = new_cert(identity_keypair, gossip_host)?;
+    let cert_chain_pem_parts: Vec<Pem> = cert_chain
+        .clone()
+        .into_iter()
+        .map(|cert| Pem {
+            tag: "CERTIFICATE".to_string(),
+            contents: cert.0,
+        })
+        .collect();
+    let cert_chain_pem = pem::encode_many(&cert_chain_pem_parts);
+
+    let mut server_config = ServerConfig::with_single_cert(cert_chain, priv_key)?;
+    let config = Arc::get_mut(&mut server_config.transport).unwrap();
+    config.max_concurrent_uni_streams(1_u8.into());
+    config.stream_receive_window(quinn::VarInt::from_u64(PACKET_DATA_SIZE as u64).unwrap());
```

Comment:
For safety, should we also set  `config.receive_window` to `MAX_CONCURRENT_UNI_STREAMS 
 * PACKET_DATA_SIZE`, `https://docs.rs/quinn/latest/quinn/struct.TransportConfig.html#method.receive_window`, so any changes to the default for this value doesn't rug us